providers/cloudns/cloudnsProvider.go

```diff
@@ -340,13 +341,33 @@ func toRc(domain string, r *domainRecord) *models.RecordConfig {
 	case "CLOUD_WR":
 		rc.Type = "WR"
 		rc.SetTarget(r.Target)
+	case "NAPTR":
+		rc.SetTargetNAPTRStrings(r.NaptrOrder, r.NaptrPref, r.NaptrFlag, r.NaptrParams, r.NaptrRegexp, r.NaptrReplace+".")
+	case "LOC":
+		loc := fmt.Sprintf("%s %s %s %s %s %s %s %s %s %s %s %s",
```

Comment:
This part should be redundant now.

Reply:
I think specifically the part that strips the decimal part of altitude is redundant? The rest has to remain as I need to assemble the LOC record from the individual API parts.

Better to call `calculateLOCFields` directly, but then I would have to duplicate the string to int/float conversions as the API returns strings for everything, despite being JSON.

Reply:
> Better to call `calculateLOCFields` directly, but then I would have to duplicate the string to int/float conversions as the API returns strings for everything, despite being JSON.

🤦 

I don't know what to say. But coercing as early as possible is safest, especially when working with numbers and maths.